File: mobile/packages/expo-hardware-keyboard/android/src/main/java/expo/modules/hardwarekeyboard/ExpoHardwareKeyboardModule.kt
package expo.modules.hardwarekeyboard

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoHardwareKeyboardModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoHardwareKeyboard")
    Constants("supportsNativeFieldBoundaries" to true, "supportsPrimaryModifierSubmit" to true,
      "supportsHardwarePaste" to true)

    View(HardwareKeyboardCaptureView::class) {
      Events("onHardwareKey")
      Prop("hardwarePaste") { view: HardwareKeyboardCaptureView, enabled: Boolean ->
        view.hardwarePaste = enabled
      }
      Prop("submitWithPrimaryModifier") { view: HardwareKeyboardCaptureView, enabled: Boolean ->
        view.submitWithPrimaryModifier = enabled
      }
      Prop("nativeFieldBoundaries") { view: HardwareKeyboardCaptureView, enabled: Boolean ->
        view.nativeFieldBoundaries = enabled
      }
      Prop("mode") { view: HardwareKeyboardCaptureView, mode: String ->
        view.captureMode = mode
      }

      Prop("enabled") { view: HardwareKeyboardCaptureView, enabled: Boolean ->
        view.captureEnabled = enabled
      }
    }
  }
}

File: mobile/packages/expo-hardware-keyboard/ios/ExpoHardwareKeyboardModule.swift
import ExpoModulesCore

public class ExpoHardwareKeyboardModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoHardwareKeyboard")
    Constants(["supportsPrimaryModifierSubmit": true])

    View(HardwareKeyboardCaptureView.self) {
      Events("onHardwareKey")
      Prop("submitWithPrimaryModifier") { (view: HardwareKeyboardCaptureView, enabled: Bool) in
        view.submitWithPrimaryModifier = enabled
      }
      Prop("mode") { (view: HardwareKeyboardCaptureView, mode: String) in
        view.captureMode = mode
      }

      Prop("enabled") { (view: HardwareKeyboardCaptureView, enabled: Bool) in
        view.enabled = enabled
      }
    }
  }
}
